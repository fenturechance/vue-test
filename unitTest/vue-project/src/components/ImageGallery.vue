<template>
  <img v-for="(image, index) in images" :key="index" :src="image" />
</template>
<script>
export default {
  props: {
    images: {
      type: Array,
      required: true,
    },
  },
}
</script>
