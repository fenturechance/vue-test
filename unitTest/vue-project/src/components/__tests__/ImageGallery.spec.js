import { mount } from "@vue/test-utils"
import { describe, it, expect } from "vitest"
import ImageGallery from "@/components/ImageGallery.vue"

describe("ImageGallery", () => {
  it("renders correctly", () => {
    const mockImages = [
      "https://placehold.co/600x400",
      "https://placehold.co/600x400/orange",
      "https://placehold.co/600x400?text=Image+3",
    ]
    const wrapper = mount(ImageGallery, {
      props: { images: mockImages },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
