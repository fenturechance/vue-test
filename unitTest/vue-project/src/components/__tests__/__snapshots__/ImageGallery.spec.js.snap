// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ImageGallery > renders correctly 1`] = `
"<img src="https://placehold.co/600x400">
<img src="https://placehold.co/600x400/orange">
<img src="https://placehold.co/600x400?text=Image+3">"
`;
